crates/optimism/storage/src/chain.rs

```diff
@@ -56,14 +57,25 @@ where
     ) -> ProviderResult<Vec<<Self::Block as Block>::Body>> {
         let chain_spec = provider.chain_spec();
 
-        for (header, _) in inputs {
+        let mut bodies = Vec::with_capacity(inputs.len());
+
+        for (header, transactions) in inputs {
             // If we are past shanghai, then all blocks should have a withdrawal list,
             // even if empty
-            if !chain_spec.is_shanghai_active_at_timestamp(header.timestamp()) {
-                return Err(ProviderError::InvalidStorageOutput);
-            }
+            let withdrawals: Option<Withdrawals> =
+                if chain_spec.is_shanghai_active_at_timestamp(header.timestamp()) {
+                    Some(vec![].into())
+                } else {
+                    None
+                };
+
+            bodies.push(alloy_consensus::BlockBody::<T, H> {
+                transactions,
+                ommers: vec![],
+                withdrawals,
+            });
         }
```

Comment:
this additional vec is not required if we iter_mut and update the withdrawals in place, right?